### PR TITLE
Migrate Kafka domain service to new module system

### LINKED ISSUE #2109: Convert kafka to module
Convert kafka (src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka) to module (this is a good second issue :D)

### PR DESCRIPTION
**Pull request by fdelbrayelle**
closes #2109

### CODE REVIEW

**Comment by DamnClin** [comment]
Looks like this module is the lat user of src/main/java/tech/jhipster/lite/generator/readme so I think this package removal should be part of this PR

**Comment by DamnClin** [comment]
/!\ breaking change incomming. In https://github.com/jhipster/jhipster-lite/pull/2500 the `dependency` method become `addDependency`. I can add some method for migration phase if you want

**Comment by fdelbrayelle** [comment]
> Looks like this module is the lat user of src/main/java/tech/jhipster/lite/generator/readme so I think this package removal should be part of this PR

Yep I will remove it!

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        .dependency(groupId("org.apache.kafka"), artifactId("kafka-clients"))
+        .dependency(groupId("org.testcontainers"), artifactId("kafka"))
+        .and()
+      .files()
+        .add(SOURCE.template("kafka.yml"), toSrcMainDocker().append("kafka.yml"))
+        .add(SOURCE.template("KafkaTestContainerExtension.java"), toSrcTestJava().append("/server/springboot/broker/kafka/KafkaTestContainerExtension.java"))
+        .add(SOURCE.template("KafkaPropertiesTest.java"), toSrcTestJava().append("/technical/infrastructure/config//kafka/KafkaPropertiesTest.java"))
+        .add(SOURCE.template("KafkaProperties.java"), toSrcMainJava().append("/technical/infrastructure/config/kafka/KafkaProperties.java"))
+        .add(SOURCE.template("KafkaConfiguration.java"), toSrcMainJava().append("/technical/infrastructure/config/kafka/KafkaConfiguration.java"))
+      .and()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:39]: missing spaces

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      .mandatoryReplacements()
+        .in(TEST_JAVA + "/IntegrationTest.java")
+          .add(text("import org.springframework.boot.test.context.SpringBootTest;"),
+            """
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:43]: extract value to a dedicated method for readability

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          import org.junit.jupiter.api.extension.ExtendWith;
+          import org.springframework.boot.test.context.SpringBootTest;""")
+          .add(text("public @interface"),
+            """
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:47]: extract value to a dedicated method for readability

**Comment by fdelbrayelle** [comment]
> /!\ breaking change incomming. In #2500 the `dependency` method become `addDependency`. I can add some method for migration phase if you want

Fixed here: https://github.com/jhipster/jhipster-lite/pull/2455/commits/c9da3d2055fe4df8b71e81b8654a5ef62736f531

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 1 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/619dcec72114e9fefcef227cf4fdb538bf2d2e71

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 2 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/619dcec72114e9fefcef227cf4fdb538bf2d2e71

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 3 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/619dcec72114e9fefcef227cf4fdb538bf2d2e71

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import tech.jhipster.lite.module.domain.properties.JHipsterModuleProperties;
+
+public class KafkaModuleFactory {
+
+  private static final JHipsterSource SOURCE = from("server/springboot/broker/kafka");
+  private static final String TECHNICAL_INFRASTRUCTURE_CONFIG_KAFKA = "/technical/infrastructure/config/kafka";
+  private static final String DUMMY_INFRASTRUCTURE_SECONDARY_KAFKA_PRODUCER = "dummy/infrastructure/secondary/kafka/producer";
+  private static final String DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER = "dummy/infrastructure/primary/kafka/consumer";
+  public static final String STRING_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+  public static final String STRING_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:18]: Why public

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/infrastructure/primary/rest/KafkaResourceConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package tech.jhipster.lite.generator.server.springboot.broker.kafka.infrastructure.primary.rest;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import tech.jhipster.lite.generator.server.springboot.broker.kafka.application.KafkaApplicationService;
+import tech.jhipster.lite.module.infrastructure.primary.JHipsterModuleApiDoc;
+import tech.jhipster.lite.module.infrastructure.primary.JHipsterModuleResource;
+
+@Configuration
+public class KafkaResourceConfiguration {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/infrastructure/primary/rest/KafkaResourceConfiguration.java:10]: Package private

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import tech.jhipster.lite.module.domain.JHipsterSource;
+import tech.jhipster.lite.module.domain.properties.JHipsterModuleProperties;
+
+public class KafkaModuleFactory {
+
+  private static final JHipsterSource SOURCE = from("server/springboot/broker/kafka");
+  private static final String TECHNICAL_INFRASTRUCTURE_CONFIG_KAFKA = "/technical/infrastructure/config/kafka";
+  private static final String DUMMY_INFRASTRUCTURE_SECONDARY_KAFKA_PRODUCER = "dummy/infrastructure/secondary/kafka/producer";
+  private static final String DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER = "dummy/infrastructure/primary/kafka/consumer";
+  public static final String STRING_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
```
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:17]: Auto public with idea, I didn't check it. Will fix it!

[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/8b953a481d7fa92eb50b9eb9dc51439bc360ea6b

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/infrastructure/primary/rest/KafkaResourceConfiguration.java (hunk 5 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/infrastructure/primary/rest/KafkaResourceConfiguration.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/8b953a481d7fa92eb50b9eb9dc51439bc360ea6b

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        .add(SOURCE.template("DummyProducerIT.java"), toSrcTestJava().append(properties.basePackage().path() + "/" + DUMMY_INFRASTRUCTURE_SECONDARY_KAFKA_PRODUCER + "/DummyProducerIT.java"))
+        .add(SOURCE.template("AbstractConsumer.java"), toSrcMainJava().append(properties.basePackage().path() + "/" + DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER + "/AbstractConsumer.java"))
+        .add(SOURCE.template("DummyConsumer.java"), toSrcMainJava().append(properties.basePackage().path() + "/" + DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER + "/DummyConsumer.java"))
+        .add(SOURCE.template("DummyConsumerTest.java"), toSrcTestJava().append(properties.basePackage().path() + "/" + DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER + "/DummyConsumerTest.java"))
+        .add(SOURCE.template("DummyConsumerIT.java"), toSrcTestJava().append(properties.basePackage().path() + "/" + DUMMY_INFRASTRUCTURE_PRIMARY_KAFKA_CONSUMER + "/DummyConsumerIT.java"))
+        .and();
+    //@formatter:on
+
+    return builder.build();
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:108]: you can inline

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        .set(propertyKey("kafka.consumer.'[value.deserializer]'"), propertyValue(STRING_DESERIALIZER))
+        .set(propertyKey("kafka.consumer.'[group.id]'"), propertyValue("myapp"))
+        .set(propertyKey("kafka.consumer.'[auto.offset.reset]'"), propertyValue("earliest"))
+        .set(propertyKey("kafka.producer.'[key.serializer]'"), propertyValue(STRING_SERIALIZER))
+        .set(propertyKey("kafka.producer.'[value.serializer]'"), propertyValue(STRING_SERIALIZER))
+        .set(propertyKey("kafka.polling.timeout"), propertyValue("10000"))
+        .and();
+    //@formatter:on
+
+    return builder.build();
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:73]: you can inline

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    final JHipsterModuleBuilder builder = moduleBuilder(properties)
+      .context()
+        .put("akhqDockerImage", dockerImages.get("tchiotludo/akhq").fullName())
+        .and()
+      .files()
+        .add(SOURCE.template("akhq.yml"), toSrcMainDocker().append("akhq.yml"))
+        .and();
+    //@formatter:on
+
+    return builder.build();
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java:122]: you can inline

File: src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    JHipsterModuleProperties properties = JHipsterModulesFixture
+      .propertiesBuilder(tmpDirForTest())
+      .basePackage("com.jhipster.test")
+      .projectBaseName("myapp")
+      .build();
+
+    JHipsterModule module = factory.buildModuleDummyProducerConsumer(properties);
+
+    var dummyProducerPath = "dummy/infrastructure/secondary/kafka/producer";
+    var dummyConsumerPath = "dummy/infrastructure/primary/kafka/consumer";
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java:110]: don't like too much the var here :D

File: src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+      .createFile(TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyProducerPath + "/DummyProducerIT.java")
+      .and()
+      .createFile(MAIN_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/AbstractConsumer.java")
+      .and()
+      .createFile(MAIN_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumer.java")
+      .and()
+      .createFile(TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumerTest.java")
+      .and()
+      .createFile(TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumerIT.java")
+      .and();
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java:132]: you could improve this part by doing something like:

```java
   .createFiles(
        MAIN_JAVA + "/" + properties.basePackage().path() + "/" + dummyProducerPath + "/DummyProducer.java",
        MAIN_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/AbstractConsumer.java",
        MAIN_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumer.java"
        )
      .createFiles(
        TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyProducerPath + "/DummyProducerTest.java",
        TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyProducerPath + "/DummyProducerIT.java",
        TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumerTest.java",
        TEST_JAVA + "/" + properties.basePackage().path() + "/" + dummyConsumerPath + "/DummyConsumerIT.java"
      )
```

Then, you could extract to private method for building mainJava and testJava to have be more readable here

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java (hunk 10 above)
Comment (fdelbrayelle) [reply src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java:110]: Yep but it's an authorized keyword for better readability IMO here :)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 7 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/abebdac9053fe4ae09a24dc1a783c960e235e918

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 8 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/abebdac9053fe4ae09a24dc1a783c960e235e918

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java (hunk 9 above)
Comment (fdelbrayelle) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactory.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/abebdac9053fe4ae09a24dc1a783c960e235e918

File: src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java (hunk 11 above)
Comment (fdelbrayelle) [reply src/test/java/tech/jhipster/lite/generator/server/springboot/broker/kafka/domain/KafkaModuleFactoryTest.java]: Fixed: https://github.com/jhipster/jhipster-lite/pull/2455/commits/abebdac9053fe4ae09a24dc1a783c960e235e918

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**